controllers/database_controller.go

```diff
@@ -677,6 +709,17 @@ func (r *DatabaseReconciler) getDatabaseSecret(ctx context.Context, dbcr *kciv1a
 	return secret, nil
 }
 
+func (r *DatabaseReconciler) annotateDatabaseSecret(ctx context.Context, dbcr *kciv1alpha1.Database, secret *corev1.Secret) error {
+	annotations := secret.ObjectMeta.GetAnnotations()
+	if len(annotations) == 0 {
+		annotations = make(map[string]string)
+	}
+	annotations[DbSecretAnnotation] = dbcr.Name
```

Comment:
question her would be, if we want to also support 1 secret to multi db's. Or we want related it's a new feature exclude this possibility. 

if we want to support this, we should make the annotation to a comma separated list.

what do you think @hyunysmile

Reply:
I was at this point too. 
but I am thinking if we really need to have cr name in annotation. 
can we just simply add label `"used-by": "db-operator",` like https://github.com/kloeckner-i/db-operator/blob/fa098fede1927411a31ea257ca1f8d144ed4a4f6/pkg/utils/kci/decorator.go#L69
instead of annotation.

Reply:
We discussed internally and I am fine with solution that @dabde suggested.

Reply:
@dabde @hyunysmile  I am still not sure why we have to implement the option when multiple `Database` resources refer to the single `Db Secret`. Currently, [Db Secret](https://github.com/kloeckner-i/db-operator/blob/master/docs/creatingdatabases.md#creatingdatabases) contains all necessary data related to the concrete DB:
```
...
data:
  POSTGRES_DB: << base64 encoded database name (generated by db operator) >>
  POSTGRES_PASSWORD: << base64 encoded password (generated by db operator) >>
  POSTGRES_USER: << base64 encoded user name (generated by db operator) >>
  CONNECTION_STRING: << base64 encoded database connection string >>
```

`Database` resource is only needed to keep the link to the corresponding `dbInstance` and keep DB backup settings.

There is no sense to use a single `Db Secret` for multiple `Database` resources.

This solution also might have issues with logic when `Db Secret` is created automatically. 
(I have created `Database` resource and db-operator created `Db Secret` automatically. How to reuse that `Db Secret`? Manually update `Db Secret` annotation? For what?)

Reply:
Only want avoid some breaking behavior for misuse. But if this happen, we can fix this in another Version.

Reply:
ok, will try to implement logic that works with comma-separated values, like in the example:
```
kind: Secret
metadata:
  annotations:
    db-operator/database: database1,database2,database3
```

Reply:
@dabde @hyunysmile 
I have [done](https://github.com/kloeckner-i/db-operator/pull/130/commits/b836bf046646dcb43a9a2ac2644ddfe1edc08dc5) with parsing comma-separated values

Please review this PR again.
Thanks.